astropy/table/table.py

```diff
@@ -1078,6 +1094,10 @@ def add_column(self, col, index=None):
             Column object to add.
         index : int or `None`
             Insert column before this position or at end (default)
+        rename_duplicate : bool
+            Uniquify column name if it already exist (default=False)
+        rename_warning : bool
```

Comment:
I would vote to drop the `rename_warning` keyword.  Either you want to allow renaming columns and are ready for the consequences, or not.  Well that's my opinion, I prefer to keep things as simple as possible and generally don't like warnings.